### PR TITLE
Add spacing around form nav buttons

### PR DESCRIPTION
**Pull request by Mottie**
## Description

The back and continue form navigation buttons are in close proximity (around 8px) to the "Finish this application later" link above and save-in-progress status alert below.

finish this application later in very close to the top of the form back button and the top of the save in progress alert is very close to the bottom of the back button

We should at least double this spacing.

Additionally, the space between the "Finish this application later" and "Open save-in-progress menu" links need to be doubled.

## Original issue(s)
[#49092](https://github.com/department-of-veterans-affairs/va.gov-team/issues/49092)

## Testing done

Visual testing

## Screenshots

After

Finish this application link above and with a 2 em spacing between the back & continue buttons, then a 2 em spacing to the save in progress status message - your application has been saved...

## Acceptance criteria
- [x] Add vertical spacing around form navigation buttons
- [x] Add horizontal spacing between finish this app and open save-in-progress links

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/platform/forms/save-in-progress/SaveFormLink.jsx [hunk 1]
```diff
@@ -42,7 +43,11 @@ class SaveFormLink extends React.Component {
     const appType = formConfig?.customText?.appType || APP_TYPE_DEFAULT;
 
     return (
-      <div className={this.props.children ? 'vads-u-display--inline' : ''}>
+      <div
+        className={`${
+          this.props.children ? 'vads-u-display--inline' : ''
+        }vads-u-margin-top--2`}
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveFormLink.jsx:49]: It looks like we could end up with something like `vads-u-display--inlinevads-u-margin-top--2` here if I'm reading it correctly. Also, is this necessary if we're already adding vertical margins to the `FormNavButtons`?

File: src/platform/forms/save-in-progress/SaveFormLink.jsx [hunk 2]
```diff
@@ -86,16 +92,25 @@ class SaveFormLink extends React.Component {
...
   form: PropTypes.shape({
     formId: PropTypes.string.isRequired,
     version: PropTypes.number.isRequired,
     data: PropTypes.object.isRequired,
     trackingPrefix: PropTypes.string.isRequired,
     savedStatus: PropTypes.string.isRequired,
+    submission: PropTypes.object.isRequired,
   }).isRequired,
-  user: PropTypes.object.isRequired,
+  locationPathname: PropTypes.string.isRequired,
```
Comment (bkjohnson) [inline src/platform/forms/save-in-progress/SaveFormLink.jsx:103]: Thank you for these ESLint fixes :+1:

File: src/platform/forms/save-in-progress/SaveFormLink.jsx (hunk 1 above)
Comment (Mottie) [reply src/platform/forms/save-in-progress/SaveFormLink.jsx:49]: Oops, you're right... I'll fix this in the morning. Huh, I thought that was adding a right margin. I'll fix that too.

Comment (Mottie) [reply src/platform/forms/save-in-progress/SaveFormLink.jsx]: I ended up reverting this margin-top addition. The form wrapper above it already includes a `vads-u-margin-y--2` and this addition didn't add any more spacing. We don't need to update the `FormNavButton` margins or the SiP menu left margin.

**Review by bkjohnson — APPROVED** [review]

**Pull request merged**